Changelog.md

```diff
@@ -4,6 +4,7 @@ Language Features:
 
 
 Compiler Features:
+ * Code Generator: Remove redundant overflow checks in specific for loops.
```

Comment:
Might be worth saying "specific popular for loops".

Or, if marketing speak is allowed... "most for loops"

Reply:
I agree that it wouldn't hurt if this was a little bit more concrete to give people a better idea of which loops get optimized this way.

We should also mention the new `isSimpleCounterLoop` annotation as an AST change.